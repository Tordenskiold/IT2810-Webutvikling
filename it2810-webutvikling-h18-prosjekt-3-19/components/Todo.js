import React, { Component } from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Alert,
} from 'react-native';

import { CheckBox, ListItem, Body } from 'native-base';

export default class Todo extends Component {
    constructor(props) {
      super(props);
      this.state = {
        text: this.props.text,
        checked: false
      }
    }


    checkBoxChecked = () => {
      this.props.completeMethod(this.props.text);
    };

    inspect = () => {
      this.props.inspectMethod(this.props.text);
    }

    render() {
      return (
        <View key={this.props.keyval} style={styles.todo}>
        <ListItem>
          <CheckBox style={styles.checkBox}
            center checked={this.state.checked}
            onPress={this.checkBoxChecked}
          />
          <Body>
            <Text style={styles.todoText} onPress={this.inspect}>{this.props.val.date}</Text>
            <Text style={styles.todoText} onPress={this.inspect}>{this.props.val.todo}</Text>
          </Body>
        </ListItem>
        <TouchableOpacity onPress={this.props.deleteMethod} style={styles.todoDelete}>
          <Text style={styles.todoDeleteText}>Delete</Text>
        </TouchableOpacity>
     </View>
    );
  }
}
const styles = StyleSheet.create({
  todo: {
    position: 'relative',
    paddingRight: 100,
    borderBottomWidth:2,
    borderBottomColor: 'white'
  },
  todoText: {
    paddingLeft: 20,
  },
  todoDelete: {
    position: 'absolute',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#2980b9',
    top: 10,
    bottom: 10,
    right: 10,
    zIndex: 11,
    borderRadius: 5,
    width: 50
  },
  checkBox: {
    borderColor: 'black',
  },
  todoDeleteText: {
    color: 'white'
  }
});
