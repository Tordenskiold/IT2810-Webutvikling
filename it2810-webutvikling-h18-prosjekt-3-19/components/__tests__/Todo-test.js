import React from "react";
import Todo from "../Todo";
import renderer from "react-test-renderer";

const completeMethod = jest.fn();

describe("Testing Todo-component", () => {
  const tree = renderer.create(
    <Todo val={""} completeMethod={completeMethod} />
  );

  describe("Testing rendering", () => {
    test("Todo renders correctly", () => {
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
