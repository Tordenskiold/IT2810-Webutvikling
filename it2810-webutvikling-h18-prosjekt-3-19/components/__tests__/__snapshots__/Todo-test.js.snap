// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Todo-component Testing rendering Todo renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "white",
      "borderBottomWidth": 2,
      "paddingRight": 100,
      "position": "relative",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 16,
          "paddingBottom": 13,
          "paddingRight": 16,
          "paddingTop": 13,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderBottomWidth": 1,
            "borderColor": "black",
            "borderLeftWidth": 1,
            "borderRadius": 13,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "height": 20,
            "left": 10,
            "marginLeft": -10,
            "marginRight": 10,
            "opacity": 1,
            "overflow": "hidden",
            "paddingBottom": 0,
            "paddingLeft": 3,
            "width": 20,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": null,
            "alignSelf": null,
            "flex": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={[Function]}
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={[Function]}
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2980b9",
        "borderRadius": 5,
        "bottom": 10,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 10,
        "top": 10,
        "width": 40,
        "zIndex": 11,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      X
    </Text>
  </View>
</View>
`;
